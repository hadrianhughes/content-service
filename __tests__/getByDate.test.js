const getByDate = require('../lib/getByDate');

jest.mock('../__mocks__/prismic-javascript.js');

const mockData = require('../__mock-data__/doc.json');
const mockApi = {
  query: q => new Promise ((resolve, reject) => {
    resolve(mockData);
  })
};

describe('getByTag function', () => {
  it('Should return a promise which resolves with an object when passed a year', () => {
    expect(getByDate(mockApi, 2000).resolves).toMatchSnapshot();
  });

  it('Should return a promise which resolves with an object when passed a year and month', () => {
    expect(getByDate(mockApi, 2000, 'January').resolves).toMatchSnapshot();
  });

  it('Should return a promise which resolves with an object when passed a year, month and day', () => {
    expect(getByDate(mockApi, 2000, 'January', 1).resolves).toMatchSnapshot();
  });

  it('Should return a promise which resolves with an object when passed a type with the date', () => {
    expect(getByDate(mockApi, 2000, 'January', 1, 'foo').resolves).toMatchSnapshot();
  });
});
