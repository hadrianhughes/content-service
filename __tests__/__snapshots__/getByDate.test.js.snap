// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getByTag function Should return a promise which resolves with an object when passed a type with the date 1`] = `undefined`;

exports[`getByTag function Should return a promise which resolves with an object when passed a year 1`] = `undefined`;

exports[`getByTag function Should return a promise which resolves with an object when passed a year and month 1`] = `undefined`;

exports[`getByTag function Should return a promise which resolves with an object when passed a year, month and day 1`] = `undefined`;
